packages/react-components/src/alert/src/Alert.tsx

```diff
@@ -191,42 +198,45 @@ const [
     });
 });
 
-// Dummy component for documentation purpose.
-export function AlertTemplate() {
-    // When returning null, react-docgen doesn't ignore the component.
-    return <></>;
-}
-
-AlertTemplate.propTypes = {
+export interface AlertTemplateProps {
     /**
      * A controlled show value.
      */
-    // eslint-disable-next-line react/no-unused-prop-types
-    show: bool,
+    show?: boolean;
     /**
      * Called when the dismiss button is clicked.
      * @param {SyntheticEvent} event - React's original SyntheticEvent.
      * @returns {void}
      */
-    // eslint-disable-next-line react/no-unused-prop-types
-    onDismiss: func,
+    onDismiss?(event: SyntheticEvent): void,
     /**
      * An HTML element type or a custom React element type to render as.
      */
-    // eslint-disable-next-line react/no-unused-prop-types
-    as: oneOfType([string, elementType]),
+    as?: ElementType;
     /**
      * React children.
      */
-    // eslint-disable-next-line react/no-unused-prop-types
-    children: any.isRequired
-};
+    children: ReactNode
+}
+
+// Dummy component for documentation purpose.
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export function AlertTemplate(_props: AlertTemplateProps) {
+    // When returning null, react-docgen doesn't ignore the component.
```

Comment:
oops made a mistake in my comment..

should be: // When returning null, react-docgen ignore the component.